### ISSUE TITLE
SSIM score replication 

### ISSUE THREAD

**Issue by Gogo693** [switchablenorms/DeepFashion_Try_On#52]
Hello,
I am trying to replicate your baseline and I produced results both by training it and using your checkpoints. I can reach your IS score but I am having trouble with the SSIM score (in both trained and checkpoint cases) where i reach only SSIM = 0.773 instead of 0.845. I am computing the SSIM by comparing the result images with corresponding ground truth ones and doing an average of all the comparisons.
Maybe I am evaluating in the wrong way?
Also, are person-shirt tuples randomized each time test is run? Because I obtain different results.
Thank you!

**Comment by An627** [comment]
hello, I am trying to replicate the baseline too, but i even can't reach the IS score, so, can you give me a link of the source of IS score? very  thanks.

**Comment by An627** [comment]
> Hello,
> I am trying to replicate your baseline and I produced results both by training it and using your checkpoints. I can reach your IS score but I am having trouble with the SSIM score (in both trained and checkpoint cases) where i reach only SSIM = 0.773 instead of 0.845. I am computing the SSIM by comparing the result images with corresponding ground truth ones and doing an average of all the comparisons.
> Maybe I am evaluating in the wrong way?
> Also, are person-shirt tuples randomized each time test is run? Because I obtain different results.
> Thank you!

hello, I am trying to replicate the baseline too, but i even can't reach the IS score, so, can you give me a link of the source of IS score? very  thanks.

**Comment by Amazingren** [comment]
> Hello,
> I am trying to replicate your baseline and I produced results both by training it and using your checkpoints. I can reach your IS score but I am having trouble with the SSIM score (in both trained and checkpoint cases) where i reach only SSIM = 0.773 instead of 0.845. I am computing the SSIM by comparing the result images with corresponding ground truth ones and doing an average of all the comparisons.
> Maybe I am evaluating in the wrong way?
> Also, are person-shirt tuples randomized each time test is run? Because I obtain different results.
> Thank you!

Hi, @Gogo693 @An627, I am also trying to replicate the baseline, have you find a right way for evaluating SSIM? 
I directly utilize the results images and the input images to compute SSIM then I get 0.770, but is seems not the right way. Is there any way for producing ground truth images to compare with generated results images?

**Comment by HAN-YANG-HIT** [comment]
halo, to compute the SSIM score correctly you should do as follows:
1. modify ACGPN_inference/data/aligned_dataset.py  to generate tryon results with the same clothes as model
2. use the ssim repo here: https://github.com/Po-Hsun-Su/pytorch-ssim      which is the exact ssim score we use in the paper
3. strictly follow the guidance of the repo above. You have to reshape the image into pytorch manner (NxCxHxW) and normalize the images to [0,1]
If correctly computed the ssim score should be 0.8664, which is higher than we reported since it is a better checkpoint than given in the paper.
